llvm/lib/ExecutionEngine/JITLink/EHFrameSupport.cpp

```diff
@@ -105,12 +105,18 @@ Error EHFrameSplitter::processBlock(LinkGraph &G, Block &B,
 
     // If this was the last block then there's nothing to split
     if (BlockReader.empty()) {
+      // If the debug_line section is being split, add an anonymous symbol so that it the blocks don't get dropped
+      if(!SectionName.compare("__DWARF,__debug_line"))
+        G.addAnonymousSymbol(B, 0, B.getContent().size(), false, true);
       LLVM_DEBUG(dbgs() << "      Extracted " << B << "\n");
       return Error::success();
     }
 
     uint64_t BlockSize = BlockReader.getOffset() - RecordStartOffset;
     auto &NewBlock = G.splitBlock(B, BlockSize);
+    // If the debug_line section is being split, add an anonymous symbol so that it the blocks don't get dropped
+    if(!SectionName.compare("__DWARF,__debug_line"))
```

Comment:
Is there a object format independent way of checking this, where we don't hardcode the Mach-O section name?